crates/vault-registry/src/lib.rs

```diff
@@ -223,15 +282,23 @@ decl_module! {
         #[weight = <T as Config>::WeightInfo::withdraw_collateral()]
         #[transactional]
         fn withdraw_collateral(origin, amount: DOT<T>) -> DispatchResult {
```

Comment:
Don't have this in one function. I suggest to keep the withdrawal function as is and if the vault is nominated, withdraw any collateral that is unbonded.

Second, add a function that triggers the unbonding. A vault can then watch off-chain if the collateral is unbonded so we can get rid of looping though all vaults on-chain if the amount can be unbonded.

Reply:
Fair enough. For now, should we use a flag set in the chain_spec of the `nomination` crate to enable/disable this feature? This would disallow vaults from opting in

Reply:
You mean like a bool that we can set via sudo? I think that would be great :)